From 7ad0a262d70af037c4352c96ef1c58b79aa01381 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Mon, 27 Aug 2018 13:25:51 +0900
Subject: [PATCH 305/909] drm: rcar-du: Skip EPROBE_DEFER in case of disabling
 LVDS and HDMI

commit b04ad20e3b3e73ee979c2de86fdef0996681f0c8 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

If the configuration was disabled, add code to skip the initialization.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_encoder.c | 22 ++++++++++++++++++++--
 1 file changed, 20 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_encoder.c b/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
index 6d7a5045df81..896f577d0e58 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
@@ -66,8 +66,26 @@ int rcar_du_encoder_init(struct rcar_du_device *rcdu,
 	/* Locate the DRM bridge from the encoder DT node. */
 	bridge = of_drm_find_bridge(enc_node);
 	if (!bridge) {
-		ret = -EPROBE_DEFER;
-		goto done;
+		if (output == RCAR_DU_OUTPUT_HDMI0 ||
+		    output == RCAR_DU_OUTPUT_HDMI1) {
+#if IS_ENABLED(CONFIG_DRM_RCAR_DW_HDMI)
+			ret = -EPROBE_DEFER;
+#else
+			ret = 0;
+#endif
+			goto done;
+		} else if (output == RCAR_DU_OUTPUT_LVDS0 ||
+			   output == RCAR_DU_OUTPUT_LVDS1) {
+#if IS_ENABLED(CONFIG_DRM_RCAR_LVDS)
+			ret = -EPROBE_DEFER;
+#else
+			ret = 0;
+#endif
+			goto done;
+		} else {
+			ret = -EPROBE_DEFER;
+			goto done;
+		}
 	}
 	renc->bridge = bridge;
 
-- 
2.17.1

